src/Account/Account.php

```diff
@@ -5,25 +5,18 @@
 /**
  * @author Milan Matějček
  */
-class Fio
+class Account
 {
+	/** @var Bank */
+	private $account;
 
 	/** @var string */
 	private $token;
 
-	/** @var Bank */
-	private $account;
-
-	public function __construct($token, Bank $account)
+	public function __construct($account, $token = NULL)
```

Comment:
Nejsem si jist zda se chápeme, proto česky. Objekt Bank se stará o to aby jakýkoliv účet byl validní a dovolil mi oddělit číslo účtu od kódu banky. A je jedno jestli jsou to fio účty, nebo účty na které posílám platby. Objekt Fio je cílený a nese si s sebou Bank (validní účet) a token který je potřeba k authentikaci. Možná by se mohl jmenovat FioAccount, ale Account mi přijde moc obecné.